# Differential State Analysis of scRNA-seq data

In brief, differential state analyses are for multi-sample multi-condition scRNA-seq datasets, where we follow specific cell _types_ across many samples (e.g., patients) in different experimental conditions (e.g., treatments) and therefore search for cell-specific responses by experimental condition}.  It is important to differentiate these analyses from the _cell type A versus cell type B_ differential expression (AB-DE) analyses of scRNA-seq data, which have received the most attention so far; in fact, we made a formal comparison of the wide range of AB-DE tools [here](https://www.nature.com/articles/nmeth.4612). Our baseline approach is to aggregate the cells of a particular type (e.g., from cell clustering, which we have also compared [here](https://f1000research.com/articles/7-1141/v2)) into sample-level measurements, mimicking what we have already proposed for flow and mass cytometry datasets in the [diffcyt](https://www.biorxiv.org/content/10.1101/349738v2) package. A *pseudobulk* approach has been proposed in the recent [demuxlet paper](https://www.nature.com/articles/nbt.4042). We have now formalized this into an R package called [`ddSingleCell`](https://github.com/HelenaLC/ddSingleCell), which takes multi-sample R objects of scRNA-seq data with cell type labels (from prediction or clustering), organizes the data into sample-level summaries and performs statistical modeling to find interesting changes in expression. The package is well integrated with visualization tools and connections to browser applications for interactive exploration of the results.

This repository gives the codebase for *differential state analyses* of single cell RNA-seq datasets (scRNA-seq), done under the [`workflowr`](https://github.com/jdblischak/workflowr) structure.
