<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="markrobinsonuzh" />

<meta name="date" content="2019-01-28" />

<title>02-diff-state</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">demuxlet_differential_state</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="01-preprocess.html">1. Preprocessing</a>
</li>
<li>
  <a href="02-diff-state.html">2. DS analysis</a>
</li>
<li>
  <a href="03-diff-state-visualizations.html">3. Visualizations</a>
</li>
<li>
  <a href="04-diff-state-genesets.html">4. Geneset analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">02-diff-state</h1>
<h4 class="author"><em>markrobinsonuzh</em></h4>
<h4 class="date"><em>2019-01-28</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-01-31</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190128)</code> </summary></p>
<p>The command <code>set.seed(20190128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/markrobinsonuzh/differential_state_examples/tree/6f2fa802ffb072a8d63002984bdf85fabae7f597" target="_blank">6f2fa80</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    chimera_tal1_differential_state/.DS_Store
    Ignored:    chimera_tal1_differential_state/.Rhistory
    Ignored:    chimera_tal1_differential_state/.Rproj.user/
    Ignored:    demuxlet_differential_state/.DS_Store
    Ignored:    demuxlet_differential_state/.Rhistory
    Ignored:    demuxlet_differential_state/.Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  chimera_tal1_differential_state/analysis/01-preprocess.Rmd
    Untracked:  chimera_tal1_differential_state/data/README.txt
    Untracked:  chimera_tal1_differential_state/data/corrected_pcas.rds
    Untracked:  chimera_tal1_differential_state/data/genes.tsv
    Untracked:  chimera_tal1_differential_state/data/meta.tab
    Untracked:  chimera_tal1_differential_state/data/raw_counts.mtx
    Untracked:  chimera_tal1_differential_state/data/sizefactors.tab
    Untracked:  demuxlet_differential_state/data/GSE96583_batch1.genes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSE96583_batch1.total.tsne.df.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSE96583_batch2.genes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSE96583_batch2.total.tsne.df.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSE96583_genes.txt.gz
    Untracked:  demuxlet_differential_state/data/GSM2560245_A.mat.gz
    Untracked:  demuxlet_differential_state/data/GSM2560245_barcodes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSM2560246_B.mat.gz
    Untracked:  demuxlet_differential_state/data/GSM2560246_barcodes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSM2560247_C.mat.gz
    Untracked:  demuxlet_differential_state/data/GSM2560247_barcodes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSM2560248_2.1.mtx.gz
    Untracked:  demuxlet_differential_state/data/GSM2560248_barcodes.tsv.gz
    Untracked:  demuxlet_differential_state/data/GSM2560249_2.2.mtx.gz
    Untracked:  demuxlet_differential_state/data/GSM2560249_barcodes.tsv.gz
    Untracked:  demuxlet_differential_state/output/GSE96583_diff_state_list.rds
    Untracked:  demuxlet_differential_state/output/GSE96583_diff_state_table.csv
    Untracked:  demuxlet_differential_state/output/GSE96583_diff_state_table.rds
    Untracked:  demuxlet_differential_state/output/GSE96583_geneset_data.rds
    Untracked:  demuxlet_differential_state/output/GSE96583_geneset_results.rds
    Untracked:  demuxlet_differential_state/output/GSE96583_geneset_results_FDR_5perc.csv
    Untracked:  demuxlet_differential_state/output/GSE96583_pseudobulks.rds
    Untracked:  demuxlet_differential_state/output/GSE96583_sce.rds

Unstaged changes:
    Modified:   demuxlet_differential_state/analysis/_site.yml

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/markrobinsonuzh/differential_state_examples/6f2fa802ffb072a8d63002984bdf85fabae7f597/demuxlet_differential_state/docs/02-diff-state.html" target="_blank">6f2fa80</a>
</td>
<td style="text-align:left;">
markrobinsonuzh
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/markrobinsonuzh/differential_state_examples/blob/5dbabcfb6fc8a81d3a478f43ec92b0cbb626e53e/demuxlet_differential_state/analysis/02-diff-state.Rmd" target="_blank">5dbabcf</a>
</td>
<td style="text-align:left;">
markrobinsonuzh
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
3/4 of way to demuxlet DS analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/markrobinsonuzh/differential_state_examples/5dbabcfb6fc8a81d3a478f43ec92b0cbb626e53e/demuxlet_differential_state/docs/02-diff-state.html" target="_blank">5dbabcf</a>
</td>
<td style="text-align:left;">
markrobinsonuzh
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
3/4 of way to demuxlet DS analysis
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(ddSingleCell)
  library(dplyr)
  library(SingleCellExperiment)
  library(limma)
})</code></pre>
</div>
<div id="load-rds-file-of-singlecellexperiment" class="section level2">
<h2>Load RDS file of <code>SingleCellExperiment</code></h2>
<pre class="r"><code>sce &lt;- readRDS(&quot;output/GSE96583_sce.rds&quot;)

# aggregate single-cell to pseudo-bulk counts
pb &lt;- aggregateData(sce, assay = &quot;counts&quot;, 
                    fun = &quot;sum&quot;, scale = FALSE)

# calculate expression frequencies by sample &amp; group
frq &lt;- calcExprFreqs(sce, assay = &quot;counts&quot;, th = 0)

# specfiy design &amp; contrast
ei &lt;- metadata(sce)$experiment_info
design &lt;- model.matrix(~ 0 + ei$group_id)
dimnames(design) &lt;- list(ei$sample_id, levels(ei$group_id))
contrast &lt;- makeContrasts(contrasts = &quot;stim-ctrl&quot;, levels = design)

design</code></pre>
<pre><code>         ctrl stim
ctrl101     1    0
ctrl1015    1    0
ctrl1016    1    0
ctrl1039    1    0
ctrl107     1    0
ctrl1244    1    0
ctrl1256    1    0
ctrl1488    1    0
stim101     0    1
stim1015    0    1
stim1016    0    1
stim1039    0    1
stim107     0    1
stim1244    0    1
stim1256    0    1
stim1488    0    1
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$`ei$group_id`
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code>contrast</code></pre>
<pre><code>      Contrasts
Levels stim-ctrl
  ctrl        -1
  stim         1</code></pre>
<pre class="r"><code># cluster-specifiy DE analysis
res &lt;- runDS(sce, pb, design, contrast,
             method = &quot;edgeR&quot;, min_cells = 10, verbose = TRUE)</code></pre>
<pre><code>B cells..CD14+ Monocytes..CD4 T cells..CD8 T cells..Dendritic cells..FCGR3A+ Monocytes..Megakaryocytes..NK cells..</code></pre>
<pre class="r"><code>saveRDS(res, &quot;output/GSE96583_diff_state_list.rds&quot;)
saveRDS(pb, &quot;output/GSE96583_pseudobulks.rds&quot;)


tbl &lt;- resDS(sce, res, bind = &quot;col&quot;, 
             frq = frq, cpm = TRUE, digits = 8) %&gt;%
  left_join(rowData(sce)  %&gt;% as.data.frame,
            by = c(&quot;gene&quot; = &quot;gene_id&quot;))

saveRDS(tbl, &quot;output/GSE96583_diff_state_table.rds&quot;)

head(tbl %&gt;% select(-ends_with(&quot;frq&quot;)) %&gt;% arrange(FDR))</code></pre>
<pre><code>                    gene        cluster_id     logFC   logCPM        F
1 ENSG00000169245_CXCL10           B cells  8.766900 11.18179 1659.787
2  ENSG00000187608_ISG15           B cells  5.499231 12.08496 1372.677
3  ENSG00000119917_IFIT3           B cells  6.830111 10.60548 1228.027
4  ENSG00000254709_IGLL5 FCGR3A+ Monocytes 11.176909 10.80328 1131.295
5  ENSG00000187608_ISG15 FCGR3A+ Monocytes  6.089216 14.18436 1070.733
6   ENSG00000108700_CCL8 FCGR3A+ Monocytes  8.744212 10.85828 1028.035
         PValue           FDR  contrast ctrl101__ctrl.cpm
1  0.000000e+00  0.000000e+00 stim-ctrl           0.00000
2 2.075390e-299 1.429425e-295 stim-ctrl          94.15265
3 3.476308e-268 1.596205e-264 stim-ctrl           0.00000
4 2.898263e-247 3.852371e-243 stim-ctrl           0.00000
5 3.528940e-234 2.345333e-230 stim-ctrl         200.96771
6 5.971562e-225 2.645800e-221 stim-ctrl           0.00000
  ctrl1015__ctrl.cpm ctrl1016__ctrl.cpm ctrl1039__ctrl.cpm
1           1.968362           6.339574            0.00000
2         261.792100         247.243400          276.40865
3          37.398871          50.716595           23.03405
4           0.000000           3.455003            0.00000
5         901.955480        1060.685900          804.57442
6           1.682753           0.000000            0.00000
  ctrl107__ctrl.cpm ctrl1244__ctrl.cpm ctrl1256__ctrl.cpm
1          64.73684          38.049090           3.833117
2          97.10526         139.513330         226.153880
3           0.00000           6.341515          45.997399
4          10.26670           0.000000           0.000000
5          82.13362         262.150180         734.133180
6           0.00000           0.000000          65.049775
  ctrl1488__ctrl.cpm stim101__stim.cpm stim1015__stim.cpm
1           7.149074       7626.168800        7433.489800
2         132.257860       8310.327200       10844.528000
3          10.723610       3571.432200        3421.765200
4           0.000000          2.949127           2.302308
5         216.708660      30768.243000       57228.470000
6           0.000000       2276.726100        5267.680700
  stim1016__stim.cpm stim1039__stim.cpm stim107__stim.cpm
1        2817.311600           4735.537          3316.500
2        8546.900500           7784.630          6860.851
3        3260.484300           2199.124          2772.189
4           3.478311          28855.725             0.000
5       31137.837000          32481.536         27361.902
6        2107.856300           4032.867          3114.818
  stim1244__stim.cpm stim1256__stim.cpm stim1488__stim.cpm symbol
1           5589.654           2343.490        2790.101000   &lt;NA&gt;
2           6240.342           9528.523        9422.023300   &lt;NA&gt;
3           2798.747           3066.171        3230.490800   &lt;NA&gt;
4              0.000              0.000           3.385382   &lt;NA&gt;
5          36647.016          37200.402       40577.185000   &lt;NA&gt;
6           2247.849           4963.632        5403.069200   &lt;NA&gt;</code></pre>
<pre class="r"><code>write.csv(tbl, &quot;output/GSE96583_diff_state_table.csv&quot;,
          row.names = FALSE, quote=FALSE)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] C

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bindrcpp_0.2.2              limma_3.38.3               
 [3] SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
 [5] DelayedArray_0.8.0          BiocParallel_1.16.5        
 [7] matrixStats_0.54.0          Biobase_2.42.0             
 [9] GenomicRanges_1.34.0        GenomeInfoDb_1.18.1        
[11] IRanges_2.16.0              S4Vectors_0.20.1           
[13] BiocGenerics_0.28.0         dplyr_0.7.8                
[15] ddSingleCell_0.99.5        

loaded via a namespace (and not attached):
 [1] viridis_0.5.1            dynamicTreeCut_1.63-1   
 [3] edgeR_3.24.3             splines_3.5.2           
 [5] viridisLite_0.3.0        DelayedMatrixStats_1.4.0
 [7] R.utils_2.7.0            assertthat_0.2.0        
 [9] statmod_1.4.30           vipor_0.4.5             
[11] GenomeInfoDbData_1.2.0   yaml_2.2.0              
[13] pillar_1.3.1             backports_1.1.3         
[15] lattice_0.20-38          glue_1.3.0              
[17] digest_0.6.18            RColorBrewer_1.1-2      
[19] XVector_0.22.0           colorspace_1.4-0        
[21] htmltools_0.3.6          Matrix_1.2-15           
[23] R.oo_1.22.0              plyr_1.8.4              
[25] pkgconfig_2.0.2          GetoptLong_0.1.7        
[27] zlibbioc_1.28.0          purrr_0.2.5             
[29] scales_1.0.0             HDF5Array_1.10.1        
[31] whisker_0.3-2            git2r_0.24.0            
[33] tibble_2.0.1             ggplot2_3.1.0           
[35] lazyeval_0.2.1           magrittr_1.5            
[37] crayon_1.3.4             evaluate_0.12           
[39] R.methodsS3_1.7.1        MAST_1.8.2              
[41] beeswarm_0.2.3           scater_1.10.1           
[43] tools_3.5.2              data.table_1.12.0       
[45] GlobalOptions_0.1.0      ComplexHeatmap_1.20.0   
[47] stringr_1.3.1            Rhdf5lib_1.4.2          
[49] munsell_0.5.0            locfit_1.5-9.1          
[51] compiler_3.5.2           rlang_0.3.1             
[53] rhdf5_2.26.2             grid_3.5.2              
[55] RCurl_1.95-4.11          BiocNeighbors_1.0.0     
[57] rjson_0.2.20             circlize_0.4.5          
[59] igraph_1.2.2             bitops_1.0-6            
[61] rmarkdown_1.11           gtable_0.2.0            
[63] abind_1.4-5              reshape2_1.4.3          
[65] R6_2.3.0                 gridExtra_2.3           
[67] knitr_1.21               zeallot_0.1.0           
[69] bindr_0.1.1              workflowr_1.1.1         
[71] rprojroot_1.3-2          shape_1.4.4             
[73] ggbeeswarm_0.6.0         stringi_1.2.4           
[75] Rcpp_1.0.0               scran_1.10.2            
[77] tidyselect_0.2.5         xfun_0.4                </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
